recipes/sparrow/all/conanfile.py

```diff
@@ -0,0 +1,112 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build.cppstd import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+from conan.tools.files import copy
+from conan.tools.microsoft import is_msvc
+from conan.tools.scm import Version
+import os
+
+required_conan_version = ">=2.0"
+
+
+class SparrowRecipe(ConanFile):
+    name = "sparrow"
+    description = "C++20 idiomatic APIs for the Apache Arrow Columnar Format"
+    license = "Apache-2.0"
+    author = "Man Group"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/man-group/sparrow"
+    topics = ("arrow", "apache arrow", "columnar format", "dataframe")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "CMakeDeps"
+    exports_sources = (
+        "include/*",
+        "LICENSE",
+        "src/*",
+        "cmake/*",
+        "docs/*",
+        "CMakeLists.txt",
+        "sparrowConfig.cmake.in",
+    )
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "use_date_polyfill": [True, False],
+    }
+    default_options = {"shared": False, "fPIC": True, "use_date_polyfill": False}
+
+    def requirements(self):
+        if self.options.get_safe("use_date_polyfill"):
+            self.requires("date/3.0.3")
+
+    @property
+    def _min_cppstd(self):
+        return 20
+
+    @property
+    def _compilers_minimum_version(self):
+        return {"apple-clang": "16", "clang": "18", "gcc": "12", "msvc": "194"}
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, self._min_cppstd)
+
+        minimum_version = self._compilers_minimum_version.get(
+            str(self.settings.compiler), False
+        )
+        if (
+            minimum_version
+            and Version(self.settings.compiler.version) < minimum_version
+        ):
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
```

Comment:
Can probably skip the compiler version checks since `check_min_cppstd` is always used on Conan v2.